source/behaviors/autosave/autosave.ts

```diff
@@ -31,34 +28,31 @@ export class AutosaveDirective implements AfterViewInit {
 	}
 
 	ngOnDestroy(): void {
-		this.autosaveCancel$.next();
+		if (this.timer) {
+			clearTimeout(this.timer);
+		}
 	}
 
 	setDebounce = (): void => {
-		if (this.canAutosave()) {
-			this.autosaveCancel$.next();
-			this.autosaveStart$.debounceTime(DEFAULT_AUTOSAVE_DEBOUNCE).takeUntil(this.autosaveCancel$).subscribe(() => this.autosave());
-			this.autosaveStart$.next();
-		} else {
-			this.autosaveCancel$.next();
+		if (!this.timer && this.form.dirty && (this.saveWhenInvalid || this.form.validate())) {
+			this.timer = setTimeout(this.autosave, DEFAULT_AUTOSAVE_DEBOUNCE)
 		}
 	}
 
 	resetDebounce(): void {
-		if (this.canAutosave()) {
-			this.autosaveStart$.next();
+		if (this.timer) {
+			clearTimeout(this.timer);
+			this.timer = null;
+			this.setDebounce();
 		}
 	}
 
 	autosave = (): void => {
-		if (!this.canAutosave()) {
```

Comment:
We still want this. This was another thing that got corrected/added with the previous PR